osu.Game/Overlays/Settings/Sections/Input/TabletAreaSelection.cs

```diff
@@ -169,12 +174,24 @@ private void checkBounds()
             if (tablet.Value == null)
                 return;
 
-            var usableSsdq = usableAreaContainer.ScreenSpaceDrawQuad;
+            // All of this manual logic is just to get around floating point issues when doing a contains check on the screen quads.
+            // This is best effort, as it's only used for display purposes. If we need for anything more, manual math on the raw values should be preferred.
+            var containerQuad = tabletContainer.ScreenSpaceDrawQuad.AABBFloat.Inflate(1);
+            var usableAreaQuad = Quad.FromRectangle(usableAreaContainer.ScreenSpaceDrawQuad.AABBFloat);
+
+            var matrix = Matrix3.Identity;
+            MatrixExtensions.TranslateFromLeft(ref matrix, usableAreaQuad.Centre);
+            MatrixExtensions.RotateFromLeft(ref matrix, MathUtils.DegreesToRadians(rotation.Value));
+            MatrixExtensions.TranslateFromLeft(ref matrix, -usableAreaQuad.Centre);
```

Comment:
Out of curiosity, was there a particular reason why you used the `FromLeft` functions here? I was a little confused because the order of operations seemed backward (as I - seemingly correctly - interpreted that what this was doing is moving the quad so that its centre is at (0,0), then rotating it and moving the centre back where it was), but I got there in the end and understood that these calls were doing the transformation composition by multiplying *from the left*. But using `FromRight` with the order swapped is just as valid and is maybe a smidge more readable.

Either way this looks to work now, so I'm not really fussed either way. :+1:

Reply:
I was just basing this off what we already did with `DrawNode` manipulation. No real reason.